### PR TITLE
Visual polish: animated wordmark, install script, interactive skill wizard

### PR DESCRIPTION
**Pull request by jeremy**
## Summary

- **Braille mountain logo** with fixed brand yellow (`#e8a217`), independent of theme
- **Bottom-up color sweep animation** — ghost silhouette appears, then a gradient wave paints upward through warm amber stages to final yellow; "Basecamp" types out to the right
- **Install script** — line-by-line logo reveal in yellow, colored step/success/error output, NO_COLOR and terminal width guards
- **Animated wordmark** on bare `basecamp` invocation and `basecamp setup` wizard (TTY only)
- **Interactive skill wizard** — `basecamp skill` shows a multi-agent picker (Agents shared, Claude Code, OpenCode, Codex, custom path) on TTY, falls back to printing skill content when piped
- **Escape to cancel** in all TUI form widgets (Select, Confirm, Input)
- **README** braille art code fence

## Test plan

- [x] `make build && ./bin/basecamp` — animated logo + quick-start on TTY
- [x] `./bin/basecamp setup` — animated logo in setup wizard
- [x] `./bin/basecamp skill` — interactive picker, escape cancels
- [x] `./bin/basecamp skill | head` — prints skill content (non-interactive)
- [x] `bash scripts/install.sh` — banner animation with "Basecamp" to the right
- [x] `make test` — all unit tests pass
- [x] `make test-e2e` — all e2e tests pass

---
## Summary by cubic
Polished the CLI’s brand and onboarding with a pluggable animated braille wordmark, a friendlier install banner, and a TTY‑only interactive skill wizard with Esc‑to‑cancel. Animations run only on TTY; piped/JSON/agent output is unchanged, output format is preserved, and we only reroute output when the animation actually runs to avoid ANSI leaks.

- **New Features**
  - Animated braille wordmark: bottom‑up warm trails to fixed brand yellow (#e8a217), then letter‑by‑letter “Basecamp”; pluggable strategies (outline‑in default; warnsdorff, radial, scanline via BASECAMP_ANIM); static on non‑TTY or NO_COLOR.
  - Async render: shown on bare `basecamp` and during `setup`; AnimWriter keeps output below the logo, and the wizard waits before prompts; setup shows the rich checklist only in interactive mode—machine output returns the JSON envelope; preserves the effective output format (e.g. --md).
  - Install script banner: TTY‑only line‑by‑line logo reveal, then type‑out; colored step/success/error; respects NO_COLOR (no cursor escapes) and narrow terminals; uses the correct binary on Windows.
  - Skill wizard (TTY only): picker for Agents (shared), Claude Code (global/project), OpenCode (global/project), Codex; supports custom path with directory normalization to basecamp/SKILL.md, ~ expansion, overwrite confirm; writes to the chosen path and `~/.agents/skills/basecamp/SKILL.md`; non‑interactive prints SKILL.md.
  - Escape‑to‑cancel in all TUI forms (Select, MultiSelect, Confirm, Input, TextArea).

- **Bug Fixes**
  - Async animation: AnimWriter now counts visual rows (ANSI‑aware, soft‑wrap) and handles exact‑fit deferred wraps to avoid repainting over output in narrow terminals.
  - Only reroute app output when the animation actually starts; static fallback keeps the original writer to prevent ANSI styles from leaking to non‑TTY destinations.
  - Unknown BASECAMP_ANIM values fall back to the default strategy instead of disabling animation; updated doc comment to match behavior.
  - Wizard success output formatting fixed (indentation/spacing).
  - Lint and module tidy: spelling/unused params fixes; move bubbles v1.0.0 to a direct dependency in go.mod.
  - Added tests for wrapping, ANSI handling, deferred‑wrap edge cases, AnimWriter accounting, strategy fallback, and valid strategy passthrough.

<sup>Written for commit 7e3cc7f728428edd847e907b5f8ba41f4a32fbff. Summary will update on new commits.</sup>

### CODE REVIEW

File: internal/commands/wizard.go [hunk 1]
```diff
@@ -283,16 +283,21 @@ func claudeManualInstallHint(styles *tui.Styles) (string, string) {
...
+	fmt.Fprintln(w)
+
 	body := fmt.Sprintf(
 		"%s\n\n%s\n\n%s",
 		styles.Body.Render(fmt.Sprintf("The command-line interface for Basecamp (v%s).", version.Version)),
 		styles.Body.Render("Let's get you set up. This will only take a moment."),
 		styles.Muted.Render("Press Enter to continue, or Ctrl+C to exit."),
 	)
-	return tui.Note(title, body)
+	return tui.Note("", body)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/wizard.go]: Fixed — showWelcome now takes `io.Writer` parameter and all wizard functions use `cmd.OutOrStdout()`.

File: internal/commands/quickstart.go [hunk 2]
```diff
@@ -65,6 +66,13 @@ func RunQuickStartDefault(cmd *cobra.Command, args []string) error {
 func runQuickStart(cmd *cobra.Command, args []string) error {
 	app := appctx.FromContext(cmd.Context())
 
+	// Show animated wordmark on interactive TTY (not JSON/agent/piped)
+	if app != nil && app.IsInteractive() {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/quickstart.go]: Fixed — added `!app.IsMachineOutput()` guard.

File: internal/commands/skill.go [hunk 3]
```diff
@@ -4,33 +4,194 @@ import (
...
-				return fmt.Errorf("reading embedded skill: %w", err)
+			var app *appctx.App
+			if ctx := cmd.Context(); ctx != nil {
+				app = appctx.FromContext(ctx)
 			}
-			_, err = fmt.Fprint(cmd.OutOrStdout(), string(data))
-			return err
+
+			// Non-interactive: print skill content (piped, --json, --agent)
+			if app == nil || !app.IsInteractive() {
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/skill.go]: Fixed — added `app.IsMachineOutput()` check.

File: internal/commands/quickstart.go [hunk 4]
```diff
@@ -65,6 +66,13 @@ func RunQuickStartDefault(cmd *cobra.Command, args []string) error {
...
+		styles := tui.NewStylesWithTheme(tui.ResolveTheme(tui.DetectDark()))
+		tui.AnimateWordmark(cmd.OutOrStdout(), styles.Theme())
+		fmt.Fprintln(cmd.OutOrStdout())
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/quickstart.go]: Fixed — added `!app.IsMachineOutput()` guard so wordmark animation is suppressed for config-driven machine formats (json/quiet).

File: internal/commands/skill.go [hunk 5]
```diff
@@ -4,33 +4,196 @@ import (
...
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/skill.go]: Fixed — added `app.IsMachineOutput()` check so skill wizard falls back to printing content for config-driven machine output.

File: internal/commands/skill.go [hunk 6]
```diff
@@ -4,33 +4,194 @@ import (
...
+				if err != nil {
+					return fmt.Errorf("reading embedded skill: %w", err)
+				}
+				_, err = fmt.Fprint(cmd.OutOrStdout(), string(data))
+				return err
+			}
+
+			// Interactive: show agent picker wizard
+			return runSkillWizard(cmd, app)
 		},
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/skill.go]: Fixed — added `app.IsMachineOutput()` check.

File: internal/tui/brand_anim.go [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+	// Pre-create a style per color stage
+	styles := make([]lipgloss.Style, len(palette))
+	for i, c := range palette {
+		styles[i] = lipgloss.NewStyle().Foreground(c)
+	}
+	textStyle := lipgloss.NewStyle().Foreground(BrandColor).Bold(true)
+
+	// Phase 1: Ghost — all rows at stage 0
+	printSweepFrame(w, lines, styles, 0, maxStage, numLines, "", textStyle, false)
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/tui/brand_anim.go]: Fixed — AnimateWordmark now detects `lipgloss.NoColor` in the theme and falls back to static `RenderWordmark` when colors are disabled.

File: internal/commands/quickstart.go [hunk 8]
```diff
@@ -65,6 +66,13 @@ func RunQuickStartDefault(cmd *cobra.Command, args []string) error {
...
+	}
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/commands/quickstart.go]: Fixed — added `!app.IsMachineOutput()` guard.

File: internal/tui/brand_anim.go [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+	// Build palette: intermediate stages + final brand color
+	stages := darkStages
+	if !theme.Dark {
+		stages = lightStages
+	}
+	palette := make([]color.Color, len(stages)+1)
+	for i, hex := range stages {
+		palette[i] = lipgloss.Color(hex)
+	}
+	palette[len(palette)-1] = BrandColor
```
[the remark this answers is not in the document]
Comment (jeremy) [reply internal/tui/brand_anim.go]: Fixed — AnimateWordmark now detects `lipgloss.NoColor` in the theme and falls back to static `RenderWordmark`.
